Disallow any dag tags longer than 100 char

TommyDew42 opened the pull request:
<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
closes #24820.

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

jwitz committed:
Update tutorial docs to include a definition of operators (#25012)

potiuk committed:
Fix markdown error in Airflow Readme (#25120)

The new markdown-lint is a bit more picky about headeers in markdown

potiuk committed:
Avoid running build ARM image twice (#24858)

In case of "merge-run" (i.e. when maintainer merges somethin
to main or v2-*-test branch is pushed, the ARM instance does not
need to be built because in such case cache is built and pushed
for ARM image.

uranusjr committed:
Bump typing-extensions and mypy for ParamSpec (#25088)

* Bump typing-extensions and mypy for ParamSpec

I want to use them in some @task signature improvements. Mypy added this
in 0.950, but let's just bump to latest since why not.

Changelog of typing-extensions is spotty before 4.0, but ParamSpec was
introduced some time before that (likely some time in 2021), and it
seems to be a reasonble minimum to bump to.

For more about ParamSpec, read PEP 612: https://peps.python.org/pep-0612/

alexott committed:
Databricks: fix test_connection implementation (#25114)

Original implementation has used the API that exists only on AWS, so it will be failing on
Azure & GCP Databricks.  This PR uses another API that is available on all Databricks
cloud platforms.

dstandish committed:
Add .../dagRuns/DR-ID/upstreamDatasetEvents endpoint (#25080)

Tells you, for a given dag run, the dataset events that are "part of" the dag run.  I.e. they were part of the collection of dataset events that contributed to the triggering of the dag run.  In practice we just query the events that occurred since the prev dag run.  We may make the relationship tighter, see https://github.com/apache/airflow/pull/24969.

chenglongyan committed:
Migrate Google example gcs_to_sftp to new design AIP-47 (#25107)

related: #22447, #22430

vincbeck committed:
Convert the batch sample dag to system tests (AIP-47) (#24448)

ecodina committed:
Added exception catching to send default email if template file raises any exception (#24943)

pastanton committed:
Makes changes to SqlToS3Operator method _fix_int_dtypes (#25083)

Convert dataframe object columns to str, to avoid errors when converting from df to parquet.

Renamed methods to remove old name:
_fix_int_dtypes -> _fix_dtypes
test_fix_int_dtypes -> test_fix_dtypes

Co-authored-by: Paul Stanton <paul.stanton@selectquote.com>

ferruzzi committed:
Convert Glue Sample DAG to System Test (#25136)

chethanuk committed:
AIP-47 - Migrate Slack DAG to new design (#25137)

blag committed:
Don't queue dagruns or create dataset events on skipped task instances (#25086)

ferruzzi committed:
Glue Job Driver logging (#25142)

Adds an optional `verbose` boolean to Glue job operators and sensors which defaults to False.  If set true, then Glue job logs will be passed through to the Airflow task logs.

chethanuk committed:
AIP-47 - Migrate redshift DAGs to new design #22438 (#24239)

ankurbajaj9 committed:
Adding Authentication to webhdfs sensor  (#25110)

potiuk committed:
Fix PR label detection in CI (#25148)

Label detection for incoming PR had few bugs:

* wrong name of output was used for Build Info
* assumption in selective checks was that PR labels are
  space separated, but they were really array formatted.
* there was a $ typo in build-images.yaml

This PR fixes all that:
* output name and typo is corrected
* we use ast.literal_eval now to parse the PR labels.

eladkal committed:
Standardize AwsLambda (#25100)

* Standardize AwsLambda
The `aws_lambda.py` is not the standard file path. The `aws_` prefix is not needed.
For the file name I used the same path as the hook.

* fix deprecated_classes.py

* update verify_providers.py

chenglongyan committed:
Migrate Google example automl_vision to new design AIP-47 (#25152)

related: #22447, #22430

potiuk committed:
Speed up Kubernetes tests ~30% on main. (#25143)

We should have enough resources - we were limiting them to 2 parallel
runs (we assumed 4 CPUs are needed for full K8S tests). However
monitoring shows that ~20/30% memory and up to 4 CPUs are used
when 2 tests are running in parallel (on a big instance) but the tests
have also some peaks that might cause more than 3 parallel tests to fail.

We currently run 5 tests so increasing the limit to 3 for self-hosted
runners should actually speed it up quite a bit (generally speaking we
had to run 2 tests in parallel twice and 5th test was run as standalone.
By increasing the limit to 3 we switch to 3x, 2x mode - which not only
will safe 30% of the time, but also has a free "slot" for one more
K8S version.

potiuk committed:
Disable provider packages building for non-main branch builds (#25056)

* Disable provider packages building for non-main branch builds

When running the builds from non-main, we do not want to use
provider packages built locally, but we want to install those from
PyPI. This is achieved by skipping the step of building providers
in case default branch is not main.

During this change it was also discovered that we do not need
AIRFLOW_FROM_CONTEXT flag to indicate whether airflow is
among those being installed from docker-context files. This is
now detected automatically and airflow is installed from Pypi
if not installed from docker context files. Providers
and airflow are automatically detected and installed if they are
present.

Installing pre-cached dependencies from GitHub is skipped in
case docker-context-files are used.

potiuk committed:
Retrieve airflow branch/constraints from env variables (#25053)

We are using the "main" breeze to build even v2-3 images and
it has to retrieve the branch and constraints not from the
Python constants but from environment variables that are
set by build-image.yaml. Otherwise "main" is used to push/pull
image and constraints.

This pr changes the retrieval in build image to retrieve
branch, constraints branch (and debian version) from env
variables if they are set.

chenglongyan committed:
Migrate Google example compute_igm to new design AIP-47 (#25132)

related: #22447, #22430

withakay committed:
Fix put-secret-value example (#25140)

To set the secret string so it is considered a list of key/value pairs we need to send a JSON object with keys and values, not an array with and object per key/value. The command as it is this will throw an error on zsh as the JSON is not escaped. After escaping the json it will update the secret as plain text and the value will simply be a JSON string. This change remedies both of these issues.

potiuk committed:
Speed up Kubernetes upgrade tests at least 2x (#25159)

We do not need to run upgrade kubernetes tests for all python
versions and all kubernetes versions. It is enough to run them
for the minimum and maximum versions. We already run all k8s tests
in main for all executors, so they "generally" work, and it
is very, very unlikely that if it works for Python 3.7, and 3.10,
it will not work for 3.8 or 3.9 (similarly for K8S version).

This will speed up the build at least 2 times.

potiuk committed:
Update description of branch creation to include the new Breeze (#25155)

* Update description of branch creation to include the new Breeze

The new Breeze has a different place where default branch and
constraints branch are kept.

When we create a new "release" branch we should update it here
as well as in the bash legacy breeze (until the bash breeze has
stil some zombies around)

bbovenzi committed:
Generate typescript types from rest API docs (#25123)

* generate api types from swagger docs, add map_index to task instance api schema

yingxuanwangxuan committed:
No grid auto-refresh for backfill dag runs (#25042)

* Update useGridData.ts

* Update useGridData.test.js

* Update useGridData.test.js

thelastmessiha committed:
Update index.rst (#25184)

This release is 3.0.1, not 3.1.0 --right?

ephraimbuddy committed:
Prefix the dags in dataset example dags with example_dataset_ (#25181)

Having dag1, dag2 etc was not telling much about the different dataset dags

potiuk committed:
Move javascript compilation to host (#25169)

Previously, in order to keep consistent development environment
we've compiled javascript in the CI image. However we can utilise
power of pre-commmit for setting the node environment for all
contributors automatically. Instead of compiling the javascript
in the image, we can compile it via pre-commit in the host.

This can be done thanks to the new python breeze which is far more
flexible and can now add execution of compilation of the javascript when
needed and using pre-commit environments.

Thanks to that, we can vastly simplify the Dockerfiles and scripts that
are used to automatically build or signal that the assets need
recompilation. We can basically assume that the assets were prepared
outside of the image building (and breeze makes sure it happens)

The changes:

* node.js is not needed in images (neither PROD build nor CI)
* no need for multiple asset compilation scripts. All is done
  via pre-commit environment with `breeze compile-www-assets``
  command
* lint checks for UI do not need the docker image any more
  (they are also based on pre-commit environment)
* no more checks/warnings when you enter the image
* start-airflow command builds the compilation before entering
* prepare-airflow-package runs asset compilation before entering
  docker airflow building.

csp33 committed:
Add revisionHistoryLimit to all deployments (#25059)

TommyDew42 committed:
raise exception when any tag length > 100

TommyDew42 committed:
an unit test for that

potiuk committed:
Fix asset compilation via setup.py (#25201)

Asset compilation via setup.py has been broken in #25169.

This PR fixes it.

pierrejeambrun committed:
Add ts types generation to static checks (#25167)

gmsantos committed:
Fix missing space for breeze build-image hint command (#25204)

chethanuk committed:
Add Tessian to list of companies using Airflow (#25206)

eladkal committed:
make MsSQL tests runnable on Python 3.8 (#25214)

mysql lib does support newer version of python https://github.com/pymssql/pymssql/pull/659

chenglongyan committed:
Migrate Google example bigquery_to_mssql to new design AIP-47 (#25174)

related: #22447, #22430

bbovenzi committed:
call updateNodeLabels after expandGroup (#25217)

eladkal committed:
make MsSQL tests runnable on Python 3.8 (2nd) (#25216)

I missed the hive provider from the commit in https://github.com/apache/airflow/pull/25214

potiuk committed:
Also compile assets in non-main (#25220)

Assets compilation when CI and PROD images are prepared should
also be run in non-main branch.

dbarrundiag committed:
SQSPublishOperator should allow sending messages to a FIFO Queue (#25171)

potiuk commented on airflow/models/dag.py:
```diff
@@ -347,6 +347,9 @@ def __init__(
     ):
         from airflow.utils.task_group import TaskGroup
 
+        if tags and any(len(tag) > 100 for tag in tags):
```
I just wonder - why 100 ? Do we know why the 100 causes the DAGs to disappear (in the original issue? ) I think having such a magic constant here is not good - it needs to come from somewhere ?

jedcunningham committed:
Show dataset readiness for the next run (#25141)

michaelosthege committed:
Include missing mention of `external_executor_id` in `sql_engine_collation_for_ids` docs (#25197)

bbovenzi committed:
Dataset details view (#25208)

* dataset details view

* Fix sorting and naming

* update table tests

* update comments

ferruzzi committed:
Convert RDS Export Sample DAG to System Test (AIP-47) (#25205)

* Convert RDS Export Sample DAG to System Test

* PR Fixes

argibbs committed:
Stop SLA callbacks gazumping other callbacks and DOS'ing the DagProcessorManager queue (#25147)

potiuk committed:
Upgrade to Pip 22.2 (#25218)

The 22.2 version of `pip` has **just** been released. Looks like
it does not break neither constraints nor "eager upgrade" builds
when we build images locally so it's safe to update to it in main.

millin committed:
Improve ElasticsearchTaskHandler (#21942)

* Improve ElasticsearchTaskHandler:
- use builtin logging.makeLogRecord instead of strange _ESJsonLogFmt
- do not re-sort already sorted logs
- apply ISO 8601 datetime format
- fixed several found bugs

potiuk committed:
Fix order of returned rows in a flaky test_outlets_dataset test (#25231)

In Postgres especially (but in generaly in all databases, if
there is no order specified in select query, the rows might
come in random order. It depends on many factors.

The test query here retrieved the dags without any order but
expected the list to be in specific order.

This PR adds ordering - it also removes side-effects of the
test by using fixture that clears the datasets before and after
the tests that rely/modify datasets - because othrwise failure of
one of the tests can create side effects that fail the other
tests (this is what happened in this case)

ferruzzi committed:
Sagemaker System Tests - Part 3 of 3 - example_sagemaker_endpoint.py (AIP-47) (#25134)

* Sagemaker System Tests - Part 3 of 3 - example_sagemaker_endpoint.py

* PR Fixes

* Fix failing static checks - unused import

potiuk reviewed (changes requested):
Yep. We need to confirm where the 100 limit comes from (and whether it's real) before we merge this one.

Commit:
Common SQLCheckOperators Various Functionality Update (#25164)

* Add batching to SQL Check Operators

Commit adds a WHERE clause to the sql statement that allows for
arbitrary batching in a given table.

* Fix bug with multiple table checks

When multiple table checks are given to the SQLTableCheckOperator
and at least one is not a fully aggregate statement, a GROUP BY
clause was previously needed. This commit updates the operator to
use the get_pandas_df() method instead of _get_first() to return a
pandas dataframe object that contains the check names and check
results from the new style of query. The new style of query uses
UNION ALL to run each test as its own SELECT statement, bypassing
the need to do a GROUP BY.

* Update test failure logic

Changed name of method from _get_failed_tests to _get_failed_checks
to better match naming, and updated logic of the method to include
an optional column param. The query in the column check operator
is removed from the failed test exception message, as it was only
ever showing the last query, instead of the relevant one(s). This is
replaced by the column, which will be more useful in debugging.

* Add table alias to SQLTableCheckOperator query

Without a table alias, the query does not run on Postgres and
other databases. The alias is arbitrary and used only for
proper query execution.

* Fix formatting error in operator

* Add batching to SQL Check Operators

Commit adds a WHERE clause to the sql statement that allows for
arbitrary batching in a given table.

* Fix bug with multiple table checks

When multiple table checks are given to the SQLTableCheckOperator
and at least one is not a fully aggregate statement, a GROUP BY
clause was previously needed. This commit updates the operator to
use the get_pandas_df() method instead of _get_first() to return a
pandas dataframe object that contains the check names and check
results from the new style of query. The new style of query uses
UNION ALL to run each test as its own SELECT statement, bypassing
the need to do a GROUP BY.

* Update test failure logic

Changed name of method from _get_failed_tests to _get_failed_checks
to better match naming, and updated logic of the method to include
an optional column param. The query in the column check operator
is removed from the failed test exception message, as it was only
ever showing the last query, instead of the relevant one(s). This is
replaced by the column, which will be more useful in debugging.

* Add table alias to SQLTableCheckOperator query

Without a table alias, the query does not run on Postgres and
other databases. The alias is arbitrary and used only for
proper query execution.

* Fix formatting error in operator

* Move alias to proper query build statement

The table alias should be in the self.sql query build statement
as that is where the table it needs to alias is defined.

* Add batching to SQL Check Operators

Commit adds a WHERE clause to the sql statement that allows for
arbitrary batching in a given table.

* Fix bug with multiple table checks

When multiple table checks are given to the SQLTableCheckOperator
and at least one is not a fully aggregate statement, a GROUP BY
clause was previously needed. This commit updates the operator to
use the get_pandas_df() method instead of _get_first() to return a
pandas dataframe object that contains the check names and check
results from the new style of query. The new style of query uses
UNION ALL to run each test as its own SELECT statement, bypassing
the need to do a GROUP BY.

* Update test failure logic

Changed name of method from _get_failed_tests to _get_failed_checks
to better match naming, and updated logic of the method to include
an optional column param. The query in the column check operator
is removed from the failed test exception message, as it was only
ever showing the last query, instead of the relevant one(s). This is
replaced by the column, which will be more useful in debugging.

* Add table alias to SQLTableCheckOperator query

Without a table alias, the query does not run on Postgres and
other databases. The alias is arbitrary and used only for
proper query execution.

* Fix formatting error in operator

* Bug fixes and updates to test and operator

Fixed bug in test where the dataframe column names did not match
the operator's expected dataframe column names. Added more info
to the SQLColumnCheckOperator's batch arg. Fixed the location of
table aliasing in SQLTableCheckOperator.

* Remove merge conflict lines

* Rename parameter batch to partition_clause

Gives a clearer name to the parameter and adds templating to
the SQLTableCheckOperator.

* Fix typo in docstring

* Reformat operator file

TommyDew42 committed:
improve test

TommyDew42 committed:
make the max length for tag 100 as a constant

TommyDew42 committed:
Merge branch 'main' of github.com:TommyDew42/airflow into disallow-dag-tag-longer-than-100-char

TommyDew42 committed:
Merge branch 'disallow-dag-tag-longer-than-100-char' of github.com:TommyDew42/airflow into disallow-dag-tag-longer-than-100-char

anselboero commented on airflow/models/dag.py:
```diff
@@ -374,6 +375,9 @@ def __init__(
         from airflow.utils.task_group import TaskGroup
 
+        if tags and any(len(tag) > TAG_MAX_LEN for tag in tags):
+            raise AirflowException("tag cannot be longer than 100 characters")
```
Sorry, short question,
should we map the Exception text to the TAG_MAX_LEN variable, too?

potiuk replied:
pls

TommyDew42 committed:
use TAG_MAX_LEN in exception message

TommyDew42 committed:
Merge branch 'main' into disallow-dag-tag-longer-than-100-char
